backend/src/nodes/nodes/stable_diffusion/clip_encode.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import torch
+
+from ...impl.stable_diffusion.types import CLIPModel, Conditioning
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import TextAreaInput
+from ...properties.inputs.stable_diffusion_inputs import CLIPModelInput
+from ...properties.outputs.stable_diffusion_outputs import ConditioningOutput
+from . import category as StableDiffusionCategory
+
+
+@NodeFactory.register("chainner:stable_diffusion:clip_encode")
+class CLIPEncodeNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = ""
+        self.inputs = [
+            CLIPModelInput(),
+            TextAreaInput("Prompt").make_optional(),
+        ]
+        self.outputs = [
+            ConditioningOutput(
+                model_type="""Conditioning {
+                arch: Input0.arch
+            }"""
+            ),
+        ]
+
+        self.category = StableDiffusionCategory
+        self.name = "CLIP Encode"
+        self.icon = "PyTorch"
+        self.sub = "Conditioning"
+
+    @torch.no_grad()
+    def run(self, clip: CLIPModel, prompt: Optional[str]) -> Conditioning:
+        prompt = prompt or ""
+        try:
+            clip.cuda()
```

Comment:
instead of doing `.cuda()`, could you do `.to(torch.device(exec_options.full_device))` like we do elsewhere? This not only ensures it goes on the right GPU but also allows things like DML to work (if someone has that manually set up)

Reply:
Done!